core/tx_pool.go

```diff
@@ -1358,6 +1371,7 @@ func (pool *TxPool) truncateQueue() {
 			}
 			drop -= size
 			queuedRateLimitMeter.Mark(int64(size))
+			delete(pool.beats, addr.address)
```

Comment:
I don't understand this. First of all, won't the `pool.removeTx(tx.Hash(),true)` internally 'fix' the `pool.beats`, if needed? And secondly, isn't it possible that you here will delete the address from beats even though not all transactions were removed?

Reply:
Yes, you are right `removeTx` will drop the `pool.beats`. 

> And secondly, isn't it possible that you here will delete the address from beats even though not all transactions were removed?

In this clause, we will drop all transactions, so in theory the queue will be empty.

But anyway i will remove it. Thanks for catching it.